.github/workflows/ci-build.yml

```diff
@@ -9,14 +9,27 @@ on:
 
 jobs:
   push_to_registry:
-    name: Push Docker image to Docker Hub
+    name: Build and Push Docker image to Docker Hub
     runs-on: ubuntu-18.04
 
     steps:
-      - name: Check out the repo
+      - name: Check out the ${{ github.event.inputs.tag }} tag to build
         uses: actions/checkout@v2
+        with:
+          ref: refs/tags/${{ github.event.inputs.tag }}
+
+      - name: Check out the ${{ github.ref }} branch for Dockerfile
+        uses: actions/checkout@v2
+        with:
+          ref: ${{github.ref}}
+          path: main_branch
+
+      - name: Copy Dockerfile from ${{github.ref}} to ${{ github.event.inputs.tag }}
+        run: |
+          cp ${{ github. workspace }}/main_branch/Dockerfile Dockerfile
+          rm -rf ${{ github. workspace }}/main_branch
 
-      - name: Push to Docker Hub
+      - name: Build Docker image ${{ github.event.inputs.tag }} and Push to DockerHub
         uses: docker/build-push-action@v1
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
```

Comment:
@hirannair Hi Hiran :) 

I see in the workflow logs that we use the password to login to Docker Hub.

`Logging in to registry` 
`WARNING! Using --password via the CLI is insecure. Use --password-stdin.`
`WARNING! Your password will be stored unencrypted in /github/home/.docker/config.json.`

It is better to use the Docker Hub access token instead of the password for security reasons. But I'm not sure if the docker build-push action supports it. But the latest version definitely supports (docker/build-push-action@v2).

Also, we have this warning in the logs:
`Warning: Input 'repository' has been deprecated with message: v2 is now available through docker/build-push-action@v2`

Do you think it is time to switch to the new version? :)

Reply:
Thanks @vensder for the review. Even though the variable is named "DOCKER_PASSWORD" ,its actually dockerhub api token we use for authentication.

Sure will have a look at the latest version of   build-push-action